src/nomad_simulations/schema_packages/workflow/base_workflows.py

```diff
@@ -0,0 +1,141 @@
+            if not self.tasks:
+                return None
+            if n_tasks is not None and len(self.tasks) != n_tasks:
+                return None
+
+            return func(self, *args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+class SimulationWorkflow(Workflow):
+    """
+    A base section used to define the workflows of a simulation with references to specific `tasks`, `inputs`, and `outputs`. The
+    normalize function checks the definition of these sections and sets the name of the workflow.
+
+    A `SimulationWorkflow` will be composed of:
+        - a `method` section containing methodological parameters used specifically during the workflow,
+        - a list of `inputs` with references to the `ModelSystem` and, optionally, `ModelMethod` input sections,
+        - a list of `outputs` with references to the `Outputs` section,
+        - a list of `tasks` containing references to the activity `Simulation` used in the workflow,
+    """
+
+    method = SubSection(
+        sub_section=BaseModelMethod.m_def,
+        description="""
+        Methodological parameters used during the workflow.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class BeyondDFTMethod(ArchiveSection):
+    """
+    An abstract section used to store references to the `ModelMethod` sections of each of the
+    archives defining the `tasks` and used to build the standard `BeyondDFT` workflow. This section needs to be
+    inherit and the method references need to be defined for each specific case (see, e.g., dft_plus_tb.py module).
+    """
+
+    pass
+
+
+class BeyondDFT(SimulationWorkflow):
+    """
+    A base section used to represent a beyond-DFT workflow and containing a `method` section which uses references
+    to the specific tasks `ModelMethod` sections.
+    """
+
+    method = SubSection(
+        sub_section=BeyondDFTMethod.m_def,
+        description="""
+        Abstract sub section used to populate the `method` of a `BeyondDFT` workflow with references
+        to the corresponding `SinglePoint` entries and their `ModelMethod` sections.
+        """,
+    )
+
+    @check_n_tasks()
+    def resolve_all_outputs(self) -> list[Outputs]:
+        """
+        Resolves all the `Outputs` sections from the `tasks` in the workflow. This is useful when
+        the workflow is composed of multiple tasks and the outputs need to be stored in a list
+        for further manipulation, e.g., to plot multiple band structures in a DFT+TB workflow.
+
+        Returns:
+            list[Outputs]: A list of all the `Outputs` sections from the `tasks`.
+        """
+        # Populate the list of outputs from the last element in `tasks`
+        all_outputs = []
+        for task in self.tasks:
+            if not task.outputs:
+                continue
+            all_outputs.append(task.outputs[-1])
```

Comment:
why only last output?

Reply:
So this method scans all tasks and gets the last output (as the one converged) of each of them, appending them in a list. Just in case we want to map to `results` or do something with the different tasks outputs.

Reply:
ah the converged output, I thought that these outputs are not necessarily results so we might include all of the outputs

Reply:
Typically we would be interested in the last `data.outputs` sections of the SinglePoints that form a BeyondDFT workflow. That's why this is a convenient method to extract those.